cpp/box/Box.h

```diff
@@ -435,6 +445,31 @@ class Box
         );
     }
 
+    //! Get mask of points that fit inside the box.
+    /*! \param points Point positions.
+        \param n_points The number of points.
+        \param cropped_mask Mask of points inside the box.
+    */
+    void crop(const vec3<float>* points, const unsigned int n_points,
+        unsigned char* cropped_mask) const
+    {
+        util::forLoopWrapper(0, n_points, [&](size_t begin, size_t end) {
+            for (size_t i = 0; i < n_points; ++i)
```

Comment:
For bonus points you could write this entire `for` loop as a single `std::transform` with a lambda function.

Reply:
Add `#include <algorithm>` at the top.
```cpp
std::transform(&points[begin], &points[end], &cropped_mask[begin],
    [this](vec3<float> point) -> unsigned char {
        return getImage(point) == vec3<int>(0, 0, 0);
    });
```
Reference: https://en.cppreference.com/w/cpp/algorithm/transform

Reply:
If I understand correctly, would this code using `std::transform` make the operation occur over a single thread rather than in parallel?

Reply:
With TBB, the full range of indices is split into sub-ranges for each thread. The outer part here:
`util::forLoopWrapper(0, n_points, [&](size_t begin, size_t end) {`
is defining that you want to go from `0` to `n_points`. Then, TBB chunks that range up into sub-ranges and sends them to each thread, which will execute from `begin` to `end`. So if you had two threads, thread 1 might go from `begin=0` to `end=n_points/2` and thread 2 might go from `begin=n_points/2` to `end=n_points`.

The "TBB Alert" I included as a separate comment was marked as resolved but this appears to still have the bug. Hopefully the explanation above helps.
> Need to be sure to use begin and end (not 0 to n_points) or else you'll have every parallel thread compute the entire range instead of just their chunk.

The `transform` I am proposing is an exact replacement for the thread-local `for` loop from `begin` to `end`.

Replace this:

```cpp
for (size_t i = 0; i < n_points; ++i)
{
	const vec3<int> image = getImage(points[i]);
	unsigned char is_in_box = 0;
	if (image == vec3<int>(0, 0, 0))
	{
		cropped_mask[i] = 1;
	}
	else
	{
		cropped_mask[i] = 0;
	}
}
```

with this:

```cpp
std::transform(&points[begin], &points[end], &cropped_mask[begin],
    [this](vec3<float> point) -> unsigned char {
        return getImage(point) == vec3<int>(0, 0, 0);
    });
```

Reply:
Ah that makes sense. 

Also I marked it as resolved because I fixed it locally but I hadn't pushed the changes yet! I wanted to see what your response was here before I pushed the changes.